coalib/io/FileFactory.py

```diff
@@ -0,0 +1,133 @@
+import os
+
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('name', 'timestamp')
+class FileFactory:
+    """
+    The ``FileFactory`` is a class to provide different views on a file.
+    It contains the following information about the file:
+
+        * The filename (absolute file path).
+        * The modification timestamp of the file.
+        * The file in raw format.
+        * The file in the form of a string UTF-8 decoded from the raw format.
+        * The lines of the file in the form of a tuple.
+
+    Two ``FileFactory`` objects are conisdered equal if they have the
+    same filename and timestamp.
+
+    To initialize a ``FileFactory`` object for a file:
+
+    >>> import os
+    >>> import tempfile
+    >>> temp = tempfile.NamedTemporaryFile(delete=False)
+    >>> temp.write(bytes('This is a test file.', 'UTF-8'))
+    20
+    >>> temp.close()
+    >>> ff = FileFactory(temp.name)
+
+    File indices start with zero.
+    To retrieve a single line:
+
+    >>> ff.get_line(0)
+    'This is a test file.'
+
+    Get all the lines in the file at once:
+
+    >>> ff.lines
+    ('This is a test file.',)
+
+    Get the file contents in raw format:
+
+    >>> ff.raw
+    b'This is a test file.'
+
+    Get the file as a string:
+
+    >>> ff.string
+    'This is a test file.'
+
+    Get the filename:
+
+    >>> ff.name == temp.name
+    True
+
+    To conveniently iterate over the lines of a file:
+
+    >>> for line in ff:
+    ...     print(line)
+    This is a test file.
+
+    >>> os.remove(ff.name)
+    """
+
+    def __init__(self, filename):
+        """
+        :param filename:
+            The filepath.
+        """
+        self._filename = os.path.abspath(filename)
+        self._timestamp = os.path.getmtime(self._filename)
+
+    def get_line(self, line):
+        """
+        :param line:
+            The line number from which the string is to be retrieved.
+            The index starts from zero.
+        :return:
+            The retrieved string at the given line number from the file.
+        """
+        try:
+            return self.lines[line]
+        except IndexError:
+            return None
```

Comment:
Do you want to catch that? Maybe let the caller catch that^^ And document that an `IndexError` is raised in case of an invalid line.